bglusman/debt_ceiling
# Travis: Use stable JRuby

This PR changes the Travis build.

It proposes:

- Use latest JRuby versions
- `JRUBY_OPTS="--debug"` for improved code coverage support
- a new Gemfile + gemspec setup to support Ruby 1.9
- Rubinius is on 3.70, which can build on this machine _and_ sadly is an allowed failure in Travis (Q: Should it be dropped from the build matrix?) [Issue in ice_nine about this](https://github.com/dkubb/ice_nine/issues/30)
- Document some of the things in the build matrix, like `haveged` and Bundler's "non-cachable"-ness

Also did:

- try and fail with RubyCritic at v3.0.0 (this worked on my machine, and seems to break < 2.3.1)
- try and fail to use JRuby's inproc setting and some other settings besides `--debug`. Went back to only `--debug`, which should enable more realistic coverage numbers
